src/plugins/lightscript.js

```diff
@@ -386,17 +386,36 @@ pp.parseNamedArrowFromCallExpression = function (node, call) {
   return this.finishNode(node, isMember ? "NamedArrowMemberExpression" : "NamedArrowExpression");
 };
 
+pp.pushBlockState = function (blockType, indentLevel) {
+  this.state.blockStack = this.state.blockStack || [];
+  this.state.blockStack.push({ blockType, indentLevel });
+};
+
+pp.matchBlockState = function(blockType, indentLevel) {
+  const stack = this.state.blockStack;
+  return (stack && stack.find( (x) => x.blockType === blockType && x.indentLevel === indentLevel ))
```

Comment:
just `return stack.some(x => x.blockType === blockType && x.indentLevel === indentLevel)` should be good